### PR TITLE
R2s fix string match problem

### PR DESCRIPTION
**Pull request by zxkjack123**
In the R2S process, we have to define the decay time in a specific manner, otherwise, there will be an error in r2s step2. For example, we define decay times like `1E3 s` or `3.0 d`, there will be an error. We have to define it like `1000 s`, `3 d`. What's more, we don't know whether there will be an error or not until the step2.

This PR is aimed to solve this problem. Float match is added in the code for better matching. Now we can define decay times in any number format as well as it meets the ALARA unit requirement. For example, decay times like `1E3 s`, `3.0 d` work well now.

### CODE REVIEW

File: pyne/alara.py [hunk 1]
```diff
@@ -843,3 +852,57 @@ def _get_subvoxel_array(mesh, cell_mats):
...
+    if unit in ('h', 'hour'):
+        return float(num) * 60 * 60
+    if unit in ('d', 'day'):
+        return float(num) * 60 * 60 * 24
+    if unit in ('w', 'week'):
+        return float(num) * 60 * 60 * 24 * 7
+    if unit in ('y', 'year'):
+        return float(num) * 60 * 60 * 24 * 7 * 52 
+    if unit in ('c', 'century'):
+        return float(num) * 60 * 60 * 24 * 7 * 52 * 100
```
Comment (scopatz) [inline pyne/alara.py]: First, these should all be `elif` after the first.

File: pyne/alara.py [hunk 2]
```diff
@@ -843,3 +852,57 @@ def _get_subvoxel_array(mesh, cell_mats):
...
     return subvoxel_array
 
+def _convert_unit_to_s(dc):
+    """
+    This function return a float number represent a time in unit of s.
+    Parameters
+    ----------
+    dc : string. Contain a num and an unit. 
+
+    return: a float number
```
Comment (scopatz) [inline pyne/alara.py]: Please use numpydoc's return section

```
Returns
-------
a float number
```

File: pyne/alara.py (hunk 1 above)
Comment (scopatz) [reply pyne/alara.py]: Second, there is a better way to do this.  First put all of your conversion factors in a global (module-level) dict.  Then, you can look up the value from this dict using either indexing or get(). I would write something like:

```python
_TO_SEC = {
  'second': 1.0,
  ...
  'century': 60 * 60 * 24 * 7 * 52 * 100.0,
}
# add abbreviations
_TO_SEC.update({k[0]: v for k, v in _TO_SEC.items()})

def _convert_unit_to_s(dc):
    num, units = dc.split()
    conv = _TO_SEC.get(units, None)
    if conv is None:
        raise ValueError('Invalid unit: {0}'.format(unit))
    return float(num) * conv
```

File: pyne/alara.py [hunk 3]
```diff
@@ -843,3 +852,57 @@ def _get_subvoxel_array(mesh, cell_mats):
...
+    """
+    This function return a float number represent a time in unit of s.
+    Parameters
+    ----------
+    dc : string. Contain a num and an unit. 
+
+    return: a float number 
+    """
+    # get num and unit
+    [num, unit] = dc.split()
```
Comment (scopatz) [inline pyne/alara.py]: I don't think you need the square brackets on the LHS.  Please remove them.

```
num, unit = dc.split()
```

**Comment by scopatz** [comment]
Thanks for putting this in!

**Comment by zxkjack123** [comment]
Update: Made some clean up according to the suggestions of @scopatz .

**Comment by scopatz** [comment]
@zxkjack123 - this doesn't seem updated other than the removal of the `[]`

**Comment by zxkjack123** [comment]
Update: Changed the unit conversion code according to the suggestion of @scopatz .

File: pyne/alara.py [hunk 4]
```diff
@@ -843,3 +852,64 @@ def _get_subvoxel_array(mesh, cell_mats):
...
+    'h': 60 * 60,
+    'hour': 60 * 60,
+    'd': 60 * 60 * 24,
+    'day': 60 * 60 * 24,
+    'w': 60 * 60 * 24 * 7,
+    'week': 60 * 60 * 24 * 7,
+    'y': 60 * 60 * 24 * 365,
+    'year': 60 * 60 * 24 * 365,
+    'c': 60 * 60 * 24 * 365 * 100,
+    'century': 60 * 60 * 24 * 365 * 100,
```
Comment (scopatz) [inline pyne/alara.py]: Years should be 365.25 days

**Comment by zxkjack123** [comment]
Update: Change year from 365 days to 365.25 days according to the comment of @scopatz .

File: pyne/alara.py (hunk 4 above)
Comment (zxkjack123) [reply pyne/alara.py]: This change has been made in the update.

File: pyne/alara.py [hunk 5]
```diff
@@ -843,3 +852,64 @@ def _get_subvoxel_array(mesh, cell_mats):
...
+    ----------
+    dc : string. Contain a num and an unit. 
+
+    Returns
+    -------
+    a float number
+    """
+    # get num and unit
+    num, unit = dc.split()
+    conv = _TO_SEC[unit]
```
Comment (scopatz) [inline pyne/alara.py]: this has to be `conv = _TO_SEC.get(unit, None)` to make the following lines work when the unit is not found.

Comment (zxkjack123) [reply pyne/alara.py]: Thanks for the comment, I have made this change in the update of this PR.

**Comment by zxkjack123** [comment]
Update: Use `_TO_SEC.get(unit, None)` to replace `_TO_SEC[unit]` according to the suggestion of @scopatz .

**Comment by scopatz** [comment]
This looks good to me! Just waiting on CI

**Pull request merged**